### PR TITLE
types: fix clustered keyword to hint regexp

### PR DESCRIPTION
**Pull request by lysu**
Signed-off-by: lysu <sulifx@gmail.com>

### What problem does this PR solve?

Problem Summary:

fix test failure in https://github.com/pingcap/ticdc/pull/1490

```
il>, ErrCount:0, SnapshotVersion:0"] [schema=clustered_index_test] [query="create database `clustered_index_test`"] [ts=423467362558148621]
[2021/03/11 02:09:52.880 +08:00] [DEBUG] [schema_storage.go:543] ["handle job: "] ["sql query"="create database `clustered_index_test`"] [job="ID:52, Type:create schema, State:done, SchemaState:public, SchemaID:51, TableID:0, RowCount:0, ArgLen:0, start time: 2021-03-11 02:09:50.032 +0800 CST, Err:<nil>, ErrCount:0, SnapshotVersion:0"]
[2021/03/11 02:09:52.881 +08:00] [DEBUG] [changefeed.go:707] ["DDL processed to make special features mysql-compatible"] [query="create database `/*T![clustered_index] clustered */ _index_test`"]
[2021/03/11 02:09:52.955 +08:00] [INFO] [mysql.go:246] ["Exec DDL succeeded"] [sql="create database `/*T![clustered_index] clustered */ _index_test`"]
```

### What is changed and how it works?

What's Changed, How it Works:

add `\b` 

### Related changes

- n/a

### Check List 

Tests 

- Unit test

Side effects

- n/a

### Release note 

- n/a

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/23247)

### CODE REVIEW

File: sessionctx/binloginfo/binloginfo_test.go [hunk 1]
```diff
@@ -614,6 +614,14 @@ func (s *testBinlogSuite) TestAddSpecialComment(c *C) {
 			"create table t1 (id int, a varchar(255), primary key (a, b) clustered);",
 			"create table t1 (id int, a varchar(255), primary key (a, b) /*T![clustered_index] clustered */ );",
 		},
+		{
+			"create table t1(id int, v int, primary key(a) clustered);",
+			"create table t1(id int, v int, primary key(a) /*T![clustered_index] clustered */ );",
+		},
+		{
+			"create table t1(id int primary key clustered, v int);", // CDC upstream => downstream never produce this
```
Comment (tangenta) [inline sessionctx/binloginfo/binloginfo_test.go:622]: Should we consider `ALTER TABLE t ADD PRIMARY KEY CLUSTERED`?

**Review by tangenta — APPROVED** [review]

File: sessionctx/binloginfo/binloginfo_test.go (hunk 1 above)
Comment (lysu) [reply sessionctx/binloginfo/binloginfo_test.go]: good catch.. and it seem cdc also will receive `create table t1(id int primary key clustered` if upstream do that..

and it seems we didn't support add pk with clustered..now

```
MySQL [test]> alter table tx add primary key(a) clustered;
Query OK, 0 rows affected (2.526 sec)

MySQL [test]> show create table tx;
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                     |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tx    | CREATE TABLE `tx` (
  `a` bigint(20) NOT NULL,
  `b` int(11) DEFAULT NULL,
  PRIMARY KEY (`a`) /*T![clustered_index] NONCLUSTERED */
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.001 sec)

```

**Comment by hi-rustin** [comment]
> /merge

If the CI test fails, you just re-trigger the test that failed and the bot will merge the PR for you after the CI passes.

**Pull request merged**